test/webfuse/tests/provider/test_client_protocol.cc

```diff
@@ -0,0 +1,167 @@
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+#include <webfuse/provider/client_protocol.h>
+#include <webfuse/provider/client_config.h>
+#include "webfuse/utils/ws_server.hpp"
+#include "webfuse/mocks/mock_provider_client.hpp"
+
+#include <cstring>
+#include <thread>
+#include <atomic>
+
+using webfuse_test::WebsocketServer;
+using webfuse_test::MockProviderClient;
+using testing::_;
+
+namespace
+{
+
+// ToDo: Refactor Me
```

Comment:
Since SetUp and TearDown are not used anymore, this class should be morphed into a utility